### DISCUSSION TITLE
How to get compressed size through writer?

### DISCUSSION THREAD

**Discussion by slhmy** [klauspost/compress#942] [Q&A] [answered]
In writer for example gzip, there is a size value indicating the uncompressed size. However, in situation like building a gzip middleware, the compressed size is also important for the Content-Length header.

I struggled for a while to find the way to get the size, but failed.

**Comment by klauspost** [comment, marked as answer]
@slhmy The gzip size is 32 bits, so it wraps at 4GB. Also you can append separate gzip files in which case you need to read each (and you can really find each without decoding)- so in practical terms it is not that useful.

It is the final 4 bytes of the gzip stored in little endian.

[gzhttp](https://github.com/klauspost/compress/tree/master/gzhttp) removes Content-Size instead, since it will do streaming writes.

**Reply by klauspost** [reply]
Alternatively you can write an `io.Writer` wrapper that counts what is being written to the Writer.

However you still have the problem that you will need to set the header *before* you start streaming any data.

**Reply by slhmy** [reply]
Yes, I'm really happy to have gzhttp as the solution. Thanks a lot!
